### ISSUE TITLE
set create_cnn.loss_func and .opt_func

### ISSUE THREAD

**Issue by Bonsen** [fastai/fastai#1346]
I define some losses like:
class FocalLoss(nn.Module):
    def __init__(self, gamma=2):
        super().__init__()
        self.gamma = gamma
        
    def forward(self, input, target):
        if not (target.size() == input.size()):
            raise ValueError("Target size ({}) must be the same as input size ({})"
                             .format(target.size(), input.size()))

        max_val = (-input).clamp(min=0)
        loss = input - input * target + max_val + \
            ((-max_val).exp() + (-input - max_val).exp()).log()

        invprobs = F.logsigmoid(-input * (target * 2.0 - 1.0))
        loss = (invprobs * self.gamma).exp() * loss
        
        return loss.sum(dim=1).mean()

bce_logit_loss =F.binary_cross_entropy_with_logits

should I use partial: 
focalLoss = partial(FocalLoss())
bce_logit_loss = partial(F.binary_cross_entropy_with_logits, pos_weight=weights)
create_cnn.loss_func = focalLoss 
create_cnn.loss_func = bce_logit_loss ???
or just use
create_cnn.loss_func = FocalLoss()
create_cnn.loss_func = F.binary_cross_entropy_with_logits ???

I define an optim:
sgd = partial(optim.SGD, momentum=0.9, weight_decay=1e-4)
create_cnn.opt_func = sgd
Is is right? What's the meaning of partial()?

**Comment by sgugger** [comment]
partial is a python function, not a fastai function. `create_cnn` is a function, you shouldn't add attributes to it. In this case, it seems you want to pass `loss_func` or `opt_func` as arguments in your call.
Again, please use the [forum](https://forums.fast.ai/) to debug your code.
